aspnetcore/fundamentals/host/generic-host.md

```diff
@@ -1,31 +1,490 @@
+
+Apps that don't handle HTTP requests can also benefit from a host's ability to manage functions such as DI, logging, configuration, and app lifetime. Such apps have one or more `IhostedService` implementations and no HTTP server implementation. For more information on using the Generic Host for apps other than ASP.NET Core, see <xref:fundamentals/host/hosted-services>.
+
+In versions of ASP.NET Core earlier than 3.0, the [Web Host](xref:fundamentals/host/web-host) is used for HTTP workloads. The Web Host is no longer recommended for web apps and remains available only for backward compatibility.
+
+## Set up a host
+
+The host is typically configured, built, and run by code in the `Program` class. The `Main` method:
+
+* Calls a `CreateHostBuilder` method to create and configure a builder object.
+* Calls `Build` and `Run` methods on the builder object.
+
+Here's *Program.cs* code for a non-HTTP workload, with a single `IHostedService` implementation added to the DI container. 
+
+```csharp
+public class Program
+{
+    public static void Main(string[] args)
+    {
+        CreateHostBuilder(args).Build().Run();
+    }
+
+    public static IHostBuilder CreateHostBuilder(string[] args) =>
+        Host.CreateDefaultBuilder(args)
+            .ConfigureServices((hostContext, services) =>
+            {
+               services.AddHostedService<Worker>();
+            });
+}
+```
+
+For an HTTP workload, the `Main` method is the same but `CreateHostBuilder` calls `ConfigureWebHostDefaults`:
+
+```csharp
+public static IHostBuilder CreateHostBuilder(string[] args) =>
+    Host.CreateDefaultBuilder(args)
+        .ConfigureWebHostDefaults(webBuilder =>
+        {
+            webBuilder.UseStartup<Startup>();
+        });
+```
+
+If the app uses Entity Framework Core, don't change the name of the `CreateHostBuilder` method. The [Entity Framework Core tools](/ef/core/miscellaneous/cli/) expect to find a `CreateHostBuilder` method that configures the host without running the app. For more information, see [Design-time DbContext Creation](/ef/core/miscellaneous/cli/dbcontext-creation).
+
+## Default builder settings 
+
+The <xref:Microsoft.Extensions.Hosting.Host.CreateDefaultBuilder*> method:
+
+* Sets the content root to the path returned by <xref:System.IO.Directory.GetCurrentDirectory*>.
+* Loads host configuration from environment variables prefixed with "DOTNET_".
+* Loads app configuration from:
+  * *appsettings.json*.
+  * *appsettings.{Environment}.json*.
+  * [Secret Manager](xref:security/app-secrets) when the app runs in the `Development` environment.
+  * Environment variables.
+  * Command-line arguments.
+* Adds the following [logging](xref:fundamentals/logging/index) providers:
+  * Console
+  * Debug
+  * EventSource
+  * EventLog (only when running on Windows)
+* Enables [scope validation](xref:fundamentals/dependency-injection#scope-validation) when the environment is Development.
```

Comment:
and dependency validation
https://github.com/aspnet/Extensions/blob/689e0d7644d5c92c1ce2f33cfc02955fa254d7f3/src/DependencyInjection/DI/src/ServiceProviderOptions.cs#L21-L25